tests/unit/test_coupling.py

```diff
@@ -83,20 +92,70 @@ def test_real_output():
 
 @pytest.mark.basic
 def test_closest_tune_approach():
+    desire = namedtuple('desire', ['err', 'isreal'])
+    map = {'teapot': desire(0, True),  # results are compared to this, from a madx-match would maybe better
+           'calaga': desire(0, True),  # same method as teapot
+           'franchi': desire(0.001, True),
+           'teapot_franchi': desire(0.0005, True),
+           'persson': desire(0.25, False),  # Not sure why it is so high here
+           'persson_alt': desire(0.25, False),
+           'hoydalsvik': desire(0.25, False),
+           'hoydalsvik_alt': desire(0.25, False),
+           }
+
     beam = 1
     df_twiss = tfs.read(INPUT / "coupling_bump" / f"twiss.lhc.b{beam:d}.coupling_bump.tfs", index=NAME)
     df = prepare_twiss_dataframe(df_twiss=df_twiss, max_order=7)
     df_cmatrix = coupling_via_cmatrix(df)
-    df_twiss[COUPLING_RDTS] = df_cmatrix[COUPLING_RDTS]
+    df_twiss[F1001] = df_cmatrix[F1001]  # ignoring F1010 in this test as it is bigger than F1010
 
-    res = dict().fromkeys(('calaga', 'franchi', 'persson', 'persson_alt'))
-    for method in res.keys():
+    res = dict().fromkeys(map.keys())
+    err = dict().fromkeys(map.keys())
+    for method, desired in map.items():
         cta = closest_tune_approach(df_twiss, method=method)
         res[method] = np.abs(np.mean(cta))[0]
+        err[method] = err_rel(res[method], res['teapot'])
+        print(f"{method}: {err} (relative error)")
+        # assert err[method] <= desired.err
+
         assert not cta.isna().any().any()
-        assert all(cta[df_cmatrix["F1001"] != 0] != 0)
-        assert all(np.abs(cta) >= 0)
-        assert err_rel(res[method], res['calaga']) < 1e-1  # ~7% for persson and persson_alt ...
+        assert all(cta[df_cmatrix[F1001] != 0] != 0)
+
+        if desired.isreal:
+            assert all(np.isreal(cta))
+
+
+@pytest.mark.basic
+def test_check_resonance_relation_with_nan(caplog):
+    df = pd.DataFrame([[1, 2, 3, 4], [2, 1, 5, 1]], index=[F1001, F1010]).T
```

Comment:
maybe just columns= and no transpose?

Reply:
but then the data is `[[1, 2], [2. 1], [3,1] [4, 1]]` and I wanted to write it column wise

Reply:
faire enough